operations/image_management/Upload_and_Register_an_Image_Recipe.md

```diff
@@ -131,3 +131,145 @@ Download and expand recipe archives from S3 and IMS. Modify and upload a recipe
     type = "s3"
     ```
 
+## Templating IMS recipes when building an IMS image
+
+IMS can optionally template the contents of an IMS recipe before building the recipe with kiwi-ng during an IMS create job. This 
+enables variables stored in the recipe to be dynamically replaced with values registered with the IMS recipe record after the 
+recipe is downloaded from S3 but before it is built by Kiwi-ng. This functionality is used by various Cray products to ensure 
+that the resulting image is built from the most correct release versions of Nexus hosted RPM repositories. 
+
+Follow the steps below to enable IMS templating within a given recipe. Note that for IMS to properly template a recipe,
+the following procedures must all be completed.
+
+### Enable templating in IMS recipe archive tgz file
+
+1. Add a file named '.ims_recipe_template.yaml' to the root of the recipe archive. The contents of this file must be valid
+   yaml containing the keyword `template_files`. The `template_files` keyword must contain a list of files to be templated
+   within the image recipe archive. 
+
+   ``` bash
```

Comment:
In general, input/output should be separated. For example:

```
ncn# ls -al
```

Example output:

```
   drwxr-xr-x 3 root root    70 Feb 22 19:05 .
   drwxr-xr-x 3 root root    41 Feb 22 17:18 ..      
   -rw-r--r-- 1 root root  2744 Feb  4 21:14 config.sh
   -rw-r--r-- 1 root root 24645 Feb  4 21:15 config.xml
   -rw-r--r-- 1 root root  1216 Feb  4 21:14 images.sh
   -rw-r--r-- 1 root root    29 Feb  4 21:14 .ims_recipe_template.yaml
   drwxr-xr-x 3 root root    18 Feb  4 21:15 root
```

Reply:
Done